pkg/virt-controller/watch/migration.go

```diff
@@ -1217,7 +1217,7 @@ func (c *MigrationController) handlePendingPodTimeout(migration *virtv1.VirtualM
 			"Migration target pod for VMI [%s/%s] is currently unschedulable.", vmi.Namespace, vmi.Name)
 		log.Log.Object(migration).Warningf("Migration target pod for VMI [%s/%s] is currently unschedulable.", vmi.Namespace, vmi.Name)
 		if secondsSpentPending >= unschedulableTimeout {
-			return c.deleteTimedOutTargetPod(migration, vmi, pod, "unschedulable pod timeout period exceeded")
+			return c.deleteTimedOutTargetPod(migration, vmi, pod, "unschedulable pod %s timeout period exceeded")
```

Comment:
Can you please hard code the string with the pod name, like `fmt.Sprintf("unschedulable pod %s timeout period exceeded", pod)`

Reply:
No, this is the template itself, `%s` should be keep, because the Pod name will be used as argument later in:

https://github.com/kubevirt/kubevirt/blob/6094537f8fe5dbbc50cea64a8c6a94d243ac8eac/pkg/virt-controller/watch/migration.go#L1159

```go
func (c *MigrationController) deleteTimedOutTargetPod(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod, reason string) error {
	// a lot of core here
	c.recorder.Eventf(migration, k8sv1.EventTypeNormal, SuccessfulDeletePodReason, reason, pod.Name) // Eventf
	return nil
}
```

Reply:
I personally find it confusing and I would also refactor the function as well. You simply pass the reason as argument. Otherwise, there is some assumptions that you know how the function works and that the template requires the pod name as string.

Reply:
> I personally find it confusing and I would also refactor the function as well. You simply pass the reason as argument. Otherwise, there is some assumptions that you know how the function works and that the template requires the pod name as string.

Yes, the pass the string template (`reason`) as argument is odd, but there are multiple caller:

https://github.com/kubevirt/kubevirt/blob/0ee4af60311200b45f36ab91c122598d219ec6e6/pkg/virt-controller/watch/migration.go#L1211-L1232

```go
	if isPodPendingUnschedulable(pod) {
		c.alertIfHostModelIsUnschedulable(vmi, pod)
		c.recorder.Eventf(
			migration,
			k8sv1.EventTypeWarning,
			MigrationTargetPodUnschedulable,
			"Migration target pod for VMI [%s/%s] is currently unschedulable.", vmi.Namespace, vmi.Name)
		log.Log.Object(migration).Warningf("Migration target pod for VMI [%s/%s] is currently unschedulable.", vmi.Namespace, vmi.Name)
		if secondsSpentPending >= unschedulableTimeout {
			return c.deleteTimedOutTargetPod(migration, vmi, pod, "unschedulable pod %s timeout period exceeded")
		} else {
			// Make sure we check this again after some time
			c.Queue.AddAfter(migrationKey, time.Second*time.Duration(unschedulableTimeout-secondsSpentPending))
		}
	}

	if secondsSpentPending >= catchAllTimeout {
		return c.deleteTimedOutTargetPod(migration, vmi, pod, "pending pod %s timeout period exceeded")
	} else {
		// Make sure we check this again after some time
		c.Queue.AddAfter(migrationKey, time.Second*time.Duration(catchAllTimeout-secondsSpentPending))
	}
```

Reply:
+1 for refactoring the function. `Eventf` at least uses `args ...interface{}` which allows to ignore it. I'd go with passing a formatted reason instead of a template.

Reply:
> +1 for refactoring the function. `Eventf` at least uses `args ...interface{}` which allows to ignore it. I'd go with passing a formatted reason instead of a template.

OK, let's refactor it.